base/tuple.jl

```diff
@@ -116,6 +116,25 @@ iterate_and_index(x) = (destruct_iterate, select_first)
 # to make inference's life easier.
 iterate_and_index(::Nothing) = throw(MethodError(iterate, (nothing,)))
 
+function _rest(itr, state, _)
+    @_inline_meta
+    state === () && return rest(itr)
```

Comment:
I think this case can be handled entirely in the front end; it only occurs for `xs... = itr`?

Reply:
Hmm, so are you saying we skip the whole `iterate_and_index` entirely in favor of plain `iterate` in this case?

Reply:
Ah sorry, I misunderstood! So we should just produce a call to `Base.rest(itr)` during lowering in that case?